### PR TITLE
Pass RenderSurvey API parameter data & Added telemetry events/Localization

### PR DESCRIPTION
**Pull request by BidishaMS**
Pass RenderSurvey API parameter data . To render NPS Survey we need following data and it is retrieved accordingly

1) TenantId - QueryParams passed from Studio
2) UsedId- Using jwt-decode of access token
3) EnvId - _Queryparams passed from Studio
4) Geo - Queryparams passed from Studio
5) Culture - Vscode locale
6) ProductVersion - Process build name
7) URLReferer - URL of the browser
8) Devicetype - Mobile/Tablet/Desktop

Added telemetry events
Localized all contents

### CODE REVIEW

File: src/web/client/services/NPSService.ts [hunk 1]
```diff
@@ -19,21 +21,29 @@ export class NPSService{
...
+                const requestInitPost: RequestInit = {
+                    method: httpMethod.POST,
+                    body:'{}',
+                    headers:NPSService.getCesHeader(accessToken)
+                };
+                const requestSentAtTime = new Date().getTime();
+                const response = await fetch(apiEndpoint, requestInitPost);
+                const result = await response?.json();
+                if( result?.eligibility){
+                    powerPlatformExtensionContext.telemetry.sendAPISuccessTelemetry(telemetryEventNames.NPS_USER_ELIGIBLE, "NPS Api",httpMethod.POST,new Date().getTime() - requestSentAtTime);
```
Comment (tyaginidhi) [inline src/web/client/services/NPSService.ts:42]: This is wrong. All you need to do it  WebExtensionContext.telemetry.sendAPISuccessTelemetry

File: src/web/client/services/NPSService.ts [hunk 2]
```diff
@@ -19,21 +21,29 @@ export class NPSService{
...
+                    WebExtensionContext.setNPSEligibility(true);
+                }
+        }catch(error){
+            powerPlatformExtensionContext.telemetry.sendErrorTelemetry(telemetryEventNames.NPS_API_FAILED, (error as Error)?.message);
```
Comment (tyaginidhi) [inline src/web/client/services/NPSService.ts:46]: same here

File: src/web/client/utilities/deviceType.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  export function isMobileDevice(): boolean {
+    const mobileRE = /(android|bb\d+|meego).+mobile|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|mobile.+firefox|netfront|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|series[46]0|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino/i;
+  
+    return mobileRE.test(navigator.userAgent);
+  }
+  
+  /**
+   * Get a flag indicating whether or not the current OS is an iOS
+   */
+  export function isIOS(): boolean {
```
Comment (tyaginidhi) [inline src/web/client/utilities/deviceType.ts:29]: This seems a bit hacky. So, this info is not available in vscode api env namespace? Let confirm once.

**Review by tyaginidhi — CHANGES_REQUESTED** [review]

File: src/web/client/services/NPSService.ts (hunk 1 above)
Comment (BidishaMS) [reply src/web/client/services/NPSService.ts]: fixed

File: src/web/client/services/NPSService.ts (hunk 2 above)
Comment (BidishaMS) [reply src/web/client/services/NPSService.ts]: fixed

File: src/web/client/utilities/deviceType.ts (hunk 3 above)
Comment (BidishaMS) [reply src/web/client/utilities/deviceType.ts]: There is no vscode api to detect tablet/mobile. Moreover it is not hacky this is a standard regex taken from detectmobilebrowsers.com(http://detectmobilebrowsers.com/))

**Review by tyaginidhi — APPROVED** [review]

**Pull request merged**